### DISCUSSION TITLE
Problem with Sign Up MFA

### DISCUSSION THREAD

**Discussion by jredondo18** [neos-sdi/adfsmfa#254] [Q&A] [answered]
Hi!

When a user is going to register an authentication application, two options appear on the first screen: Register and Login. 

If the user clicks on register, the registration process starts and key activation begins.

But if the user clicks on login, he/she accesses directly to the application, without previous registration.

Is there any way to remove the login button?

![image](https://user-images.githubusercontent.com/22500969/193786053-552ae048-52bb-4a5c-a58d-2d8b655513bd.png)

Regards!

**Comment by redhook62** [comment, marked as answer]
Hi, @jredondo18 

Yes, you are on the default template. 
This authorizes the user to bypass the MFA, however he will be reminded regularly that he must register. 
As you can see here : [Templates](https://github.com/neos-sdi/adfsmfa/wiki/05-General%20Settings#user-features-policies-templates), 
There is a set of templates that can address a large part of the cases that the administrator can evaluate.
If you want users to be forced to register without being able to bypass MFA, I think you should use the "Mixed" template, you can also look at the different possibilities again here: [Templates](https://github.com/neos-sdi/adfsmfa/wiki/05-General%20Settings#user-features-policies-templates).

regards

redhook

**Comment by jredondo18** [comment]
Thank you very much!

Another question, is it possible to modify the text in each box?

Thank you, best regards.

**Comment by redhook62** [comment]
Hi, **@jredondo18**

Yes, there are 2 ways to do it.
1) If your modifications are generic you can modify the resource files and post a "Pul Request", and if all this suits us it will be integrated into a build.

2) You can modify all the resource files concerning the UI (there is one set per language), for that refer here: [Localization](https://github.com/neos-sdi/adfsmfa/wiki/11-Localization) and follow the procedure. 
In this case you must continue to maintain your translations in the future. it is also possible to modify the emails sent as indicated in the same chapter.

regards

redhook

**Comment by jredondo18** [comment]
Hello, thank you for your reply,

Back to my first question. If I activate the "Mixed" template, now I see how the registration is mandatory, but I can not modify the options management. I don't want the user to access the options panel.

![image](https://user-images.githubusercontent.com/22500969/194061066-b2cbc005-7f3b-4643-8be9-518de21e04bb.png)

Is there any way to remove that option to the user?

Thanks again,

Regards!

**Comment by redhook62** [comment]
Hi @jredondo18 

nothing could be simpler, 
The models impose fixed choices in part. 
The easiest way for you is to use the "Custom" template, you will be able to access all the options

regards

**Reply by jredondo18** [reply]
Hi,

I have selected the custom template, however, this again authorizes the user to bypass registration.

This is my global MFA configuration:

![image](https://user-images.githubusercontent.com/22500969/194079203-610b5750-4b44-43c0-ad17-f3c27d30c722.png)

And now I see again the login button on the first registration step:

![image](https://user-images.githubusercontent.com/22500969/194079266-2c1e37d4-8c9f-482d-a6f9-2eec8a90b0b8.png)

Regards!

**Comment by redhook62** [comment]
Hi, **@jredondo18** 

Once you have defined your template in general settings (these apply to the user in general, regardless of authentication providers (TOTP, mail, Biometrics)).

![General config](https://user-images.githubusercontent.com/26826945/194091871-a47061d2-4a99-4e28-bd7a-7b7d4fa4f6cd.png)

So with the "Mixed" or "Custom" template you force users to register, they can't continue without going through this step.

Then, on each authentication provider, you will intervene to define a possibly specific behavior.
When registering the user, if the provider is not required, the user can continue without registering on this same provider.

![Providers](https://user-images.githubusercontent.com/26826945/194091508-0e345b06-90fc-46e4-a939-a6c32f09b491.png)

So you must indicate that the TOTP provider is **Required**.

regards

redhook

**Reply by jredondo18** [reply]
Sorry, but I keep seeing the login button in the registration panel.

This is my configuration:

![image](https://user-images.githubusercontent.com/22500969/194119317-9d5a31c6-3809-4db1-8d86-00c575d8b0cf.png)
![image](https://user-images.githubusercontent.com/22500969/194119345-84307539-844c-4cdb-a786-aafd437b67b5.png)

Is there something wrong? I have restarted the ADFS services, and cleared the browser cache.

Regards.

**Comment by redhook62** [comment]
Hi @jredondo18 

We just checked your configuration on an ADFS 2019, no worries, it's exactly as we told you.
![sample](https://user-images.githubusercontent.com/26826945/194121848-9b71f01a-dbb3-4c45-9813-6ff4f260b8a9.png)

Now !
- What is your version of ADFS ?
- Do you have multiple servers ?
-- If Yes ! did you open the firewalls as indicated in the documentation ?
- What is the storage mode (ADDS or SQL) ?

Thank you in advance for sending us this information.

regards

**Reply by jredondo18** [reply]
Hi!

> What is your version of ADFS ?
ADFS 2012

> Do you have multiple servers ?
No, and the firewalls is open

> What is the storage mode (ADDS or SQL) ?
ADDS

Thanks!

Regards.

**Comment by redhook62** [comment]
Hi,

Ok, we reproduce your problem on ADFS 2012R2.
On the other hand, it is probably not an ADFS 2012, because the model is not compatible and especially Windows 2012 is not supported. can you tell us exactly what the ADFS version is (2012R2, 2016, 2019 or 2022) ?

regards

**Reply by jredondo18** [reply]
Hi,

Sorry, is ADFS 2012R2.

Regards.

**Comment by redhook62** [comment]
Hi, 

new release version 3.1.2210.1. this will solve your problem.

regards

**Reply by jredondo18** [reply]
Hi @redhook62,

Thank you very much for your time,

The error has been corrected correctly! 

When I access the MFA console, and select "MFA Providers" to disable the email and biometrics provider, I get the following error:

![image](https://user-images.githubusercontent.com/22500969/194316531-853ba845-51c6-4b41-bd68-7d1ad480e3fc.png)
![image](https://user-images.githubusercontent.com/22500969/194316549-3f9ce787-14fe-438e-9b52-2002d00fdee6.png)

Do you know what it could be?

**Reply by redhook62** [reply]
Hi,

Yes, it's confirmed on ADFS 212r2 and 2016.
There is a problem with nugets packages (Microsoft does nonsense...)
We look in detail.

regards

**Reply by redhook62** [reply]
Ok, take the last build, everything is solved. 3.1.2210.1

**Reply by jredondo18** [reply]
Hi @redhook62,

I confirm that's correct. 

Thank you for your time.

Regards!
